rllib/env/tests/test_single_agent_episode.py

```diff
@@ -0,0 +1,303 @@
+    def __init__(self):
+        self.observation_space = gym.spaces.Discrete(201)
+        self.action_space = gym.spaces.Discrete(200)
+        self.t = 0
+
+    def reset(
+        self, *, seed: Optional[int] = None, options=Optional[Dict[str, Any]]
+    ) -> Tuple[ObsType, Dict[str, Any]]:
+        self.t = 0
+        return 0, {}
+
+    def step(
+        self, action: ActType
+    ) -> Tuple[ObsType, SupportsFloat, bool, bool, Dict[str, Any]]:
+        self.t += 1
+        if self.t == 200:
+            is_terminated = True
+        else:
+            is_terminated = False
+
+        return self.t, self.t, is_terminated, False, {}
+
+
+class TestSingelAgentEpisode(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_init(self):
+        # Create empty episode.
+        episode = SingleAgentEpisode()
+        # Empty episode should have a start point and count of zero.
+        self.assertTrue(episode.t_started == episode.t == 0)
+
+        # Create an episode with a specific starting point.
+        episode = SingleAgentEpisode(t_started=10)
+        self.assertTrue(episode.t == episode.t_started == 10)
+
+        # Sample 100 values and initialize episode with observations.
+        env = gym.make("CartPole-v1")
+        observations = rewards = is_terminateds = is_truncateds = []
+        init_obs, _ = env.reset()
+        observations.append(init_obs)
+        for _ in range(100):
+            action = env.action_space.sample()
+            obs, reward, is_terminated, is_truncated, _ = env.step(action)
+            observations.append(obs)
+            rewards.append(reward)
+            is_terminateds.append(is_terminated)
+            is_truncateds.append(is_truncated)
+
+        episode = SingleAgentEpisode(
+            observations=observations,
+            rewards=rewards,
+            is_terminated=is_terminateds,
+            is_truncated=is_truncated,
+        )
+        # The starting point and count should now be at `len(observations) - 1`.
+        self.assertTrue(episode.t == episode.t_started == (len(observations) - 1))
+
+    def test_add_initial_observation(self):
+        # Create empty episode.
+        episode = SingleAgentEpisode()
+        # Create environment.
+        env = gym.make("CartPole-v1")
+
+        # Add initial observations.
+        obs, _ = env.reset()
+        episode.add_initial_observation(initial_observation=obs)
+
+        # Assert that the observations are added to list.
+        self.assertTrue(len(episode.observations) == 1)
+        # Assert that the timesteps are still at zero as we have not stepped, yet.
+        self.assertTrue(episode.t == episode.t_started == 0)
+
+    def test_add_timestep(self):
+        # Set the random seed (otherwise the episode will terminate at
+        # different points in each test run).
+        np.random.seed(0)
```

Comment:
Dumb question: I though one would have to do `env.reset(seed=...)` now in gymnasium.
Does it matter, which one we do?

Reply:
I try this out. This might be the better option. Thanks for the hint @sven1977